dll/cpl/input/input_list.c

```diff
@@ -662,6 +680,73 @@ InputList_Create(VOID)
     free(pLayoutList);
 }
 
+static INT InputList_Compare(INPUT_LIST_NODE *pNode1, INPUT_LIST_NODE *pNode2)
+{
+    INT nCompare = lstrcmpiW(pNode1->pszIndicator, pNode2->pszIndicator);
+    if (nCompare != 0)
+        return nCompare;
+
+    return lstrcmpiW(pNode1->pLayout->pszName, pNode2->pLayout->pszName);
+}
+
+VOID InputList_Sort(VOID)
+{
+    INPUT_LIST_NODE *pNode, *pNext, *pPrev, *pMinimum, *pList = _InputList;
+
+    _InputList = NULL;
+
+    while (pList)
+    {
+        /* Find the minimum node */
+        pMinimum = NULL;
+        for (pNode = pList; pNode; pNode = pNext)
+        {
+            pNext = pNode->pNext;
+
+            if (pMinimum == NULL)
+            {
+                pMinimum = pNode;
+            }
+            else
+            {
+                if (InputList_Compare(pNode, pMinimum) < 0)
+                {
+                    pMinimum = pNode;
+                }
+            }
+        }
+
+        // Remove pMinimum from pList
+        pNext = pMinimum->pNext;
+        pPrev = pMinimum->pPrev;
+        if (pNext)
+            pNext->pPrev = pPrev;
+        if (pPrev)
+            pPrev->pNext = pNext;
+        else
+            pList = pNext;
+
+        // Append pMinimum to _InputList
+        if (!_InputList)
+        {
+            pMinimum->pPrev = pMinimum->pNext = NULL;
+            _InputList = pMinimum;
+        }
+        else
+        {
+            /* Find last node */
+            for (pNode = _InputList; pNode->pNext; pNode = pNode->pNext)
+            {
+                ;
```

Comment:
`NOTHING` is not defined. I defined it in this file.

Reply:
Shouldn't it be in the winnt.h? I don't remember :thinking:
Otherwise the loop could be: `pNode = _InputList; while (pNode->pNext) { pNode = pNode->pNext; }`
Anyway, I'm ok with the define as well.